utilities/network.py

```diff
@@ -1038,23 +1038,146 @@ def get_cluster_cni_type(admin_client):
+        if iface.name in pf_names or iface.pciAddress in root_devices:
+            matching_interfaces.append(iface)
+
+    if not matching_interfaces:
+        LOGGER.info(
+            f"Skipping verification for node {sriov_node_network_state.name} - "
+            f"no interfaces matching policy {policy.name} (pfNames={pf_names}, rootDevices={root_devices})"
+        )
+        return
+
+    for iface in matching_interfaces:
+        LOGGER.info(
+            f"Verifying interface {iface.name} on node {sriov_node_network_state.name}: "
+            f"numVfs={iface.numVfs} (expected: {expected_num_vfs})"
+        )
+
+        if iface.numVfs != expected_num_vfs:
+            LOGGER.error(
+                f"Interface {iface.name} on {sriov_node_network_state.name} has numVfs={iface.numVfs}, "
+                f"expected {expected_num_vfs}"
+            )
+            raise AssertionError(f"numVfs mismatch for {iface.name}: got {iface.numVfs}, expected {expected_num_vfs}")
+
+        if expected_mtu:
+            iface_mtu = getattr(iface, "mtu", None)
+            if iface_mtu and iface_mtu != expected_mtu:
+                LOGGER.error(
+                    f"Interface {iface.name} on {sriov_node_network_state.name} has mtu={iface_mtu}, "
+                    f"expected {expected_mtu}"
+                )
+                raise AssertionError(f"MTU mismatch for {iface.name}: got {iface_mtu}, expected {expected_mtu}")
+
+        LOGGER.info(f"Interface {iface.name} configuration verified successfully")
+
+
+def wait_for_ready_sriov_nodes(snns, policy=None):
+    """
+    Wait for SR-IOV nodes to be ready and verify interface configuration matches policy.
+
+    Args:
+        snns: List of SriovNetworkNodeState objects
+        policy: Optional SriovNetworkNodePolicy object to verify configuration against
+
+    Raises:
+        TimeoutExpiredError: If sync status doesn't reach SUCCEEDED state or configuration doesn't match
+        AssertionError: If interface configuration doesn't match policy
+    """
+    for sriov_node_network_state in snns:
+        LOGGER.info(f"Waiting for node {sriov_node_network_state.name} to reach SUCCEEDED status")
+        sampler = TimeoutSampler(
+            wait_timeout=1000,
+            sleep=5,
+            func=lambda snns=sriov_node_network_state: snns.instance.status.syncStatus,
+        )
+        try:
+            for sample in sampler:
+                if sample == SriovNetworkNodePolicy.Status.SUCCEEDED:
+                    LOGGER.info(f"Node {sriov_node_network_state.name} reached SUCCEEDED status")
+                    break
+        except TimeoutExpiredError:
+            current_status = sriov_node_network_state.instance.status.syncStatus
+            LOGGER.error(
+                f"Timeout waiting for node {sriov_node_network_state.name} to reach SUCCEEDED status. "
+                f"Current status: {current_status}"
+            )
+            raise
+
+    if policy:
```

Comment:
this function is too long and should be split into 2 helpers and can be called together from another fucntion